### PR TITLE
react app root path can be override by env var 'react_native_app_root'

### PR DESCRIPTION
**Pull request by chrisliu529**
This PR is to solve app build issue when node_modules is a symlink by providing an environmental variable to override the current *smart* guessing of app root path.
I met this issue when I tried to setup a shared incremental node_modules directory to speed our react-native app build speed in CI. But the build crashed in step 'bundleReleaseJsAndAssets' with error messages like:
> :app:bundleReleaseJsAndAssets
> bundle: Created ReactPackager
> uncaught error Error: NotFoundError: Cannot find entry file index.android.js in any of the roots:["/home/jenkins/shared_data"]

The build is fixed by applying this patch and adding 'export react_native_app_root=${WORKSPACE}' before './gradlew assembleRelease' in build script.

**Test plan**
1. react-native init demo # init a demo app from scratch
2. cd demo/android && ./gradlew assembleRelease # build works fine
3. mkdir ~/shared_data && mv ../node_modules ~/shared_data && cd .. && ln -s ~/shared_data/node_modules . # create symlink for node_modules in shared data directory
4.  cd demo/android && ./gradlew assembleRelease # build crashed with error message 'uncaught error Error: NotFoundError: Cannot find entry file index.android.js in any of the roots: ["..."]'
5. apply this patch to ~/shared_data/node_modules/react-native/local-cli/default.config.js
6. export react_native_app_root=${demo-app-path}
7. rebuild and it works

### CODE REVIEW

File: local-cli/default.config.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ var config = {
 };
 
 function getRoots() {
+  var root = process.env.react_native_app_root;
```
Comment (mkonicek) [inline local-cli/default.config.js]: What's the convention for env variables used elsewhere? Should it be `REACT_NATIVE_APP_ROOT`? Not sure, just checking.

**Comment by mkonicek** [comment]
cc @bestander (@martinbigio moved to a different team)

File: local-cli/default.config.js (hunk 1 above)
Comment (chrisliu529) [inline local-cli/default.config.js]: Actually I'm new to react-native community and I've no idea what its convention is. If the env var should be capitalised then  REACT_NATIVE_APP_ROOT would be better :)
And I just grep all existing 'process.env' references in current code base, indeed all env vars are capitalised. I agree REACT_NATIVE_APP_ROOT is better.

**Comment by bestander** [comment]
Does not seem to break anything and as long as it helps a common scenario LGTM.

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**